File: ServiceDesk/Models/Ticket.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ServiceDesk.Models
{
    public class Ticket
    {
        [Key]
        public int Id { get; set; }
        public string? Titulo { get; set; }
        public string? Descricao { get; set; }
        public DateTime dataAbertura { get; set; }

        public int? StatusId { get; set; }

        public int? PrioridadeId { get; set; }

        public int? CategoriaId { get; set; }

        public int? DispositivoId { get; set; }

        public int? UsuarioId { get; set; }

        public int? FuncionarioId { get; set; }

        public int? SolucaoId { get; set; }

        public int? SlaId { get; set; }

        [JsonIgnore]
        public Sla? sla { get; set; }
        public Status? status { get; set; }
        public Prioridade? prioridade { get; set; }
        public Categoria? categoria { get; set; }
        public Dispositivo? Dispositivo { get; set; }
        public Usuario? usuarioCriador { get; set; }
        public Funcionario? funcionarioResponsavel { get; set; }
        public Solucao? Solucao { get; set; }
    }
}

File: AppServiceDesk/src/app/Ticket.ts
import { Dispositivo } from "./Dispositivo";
import { Status } from "./Status";
import { Prioridade } from "./Prioridade";
import { Categoria } from "./Categoria";
import { Sla } from "./Sla";
import { Funcionario } from "./Funcionario";
import { Usuario } from "./Usuario";


export class Ticket {
    id?: number = 0;
    titulo: string = "";
    descricao: string = "";
    dataAbertura: Date = new Date();
    status: Status | undefined;
    statusId:  number = 0;
    prioridade: Prioridade | undefined;
    prioridadeId: number = 0;
    categoria: Categoria | undefined;
    categoriaId: number = 0;
    sla: Sla | undefined;
    slaId: number = 0;
    dispositivo: Dispositivo | undefined;
    dispositivoId: number = 0;
    usuario: Usuario | undefined;
    usuarioId: number = 0;
    funcionario: Funcionario | undefined;
    funcionarioId: number = 0;
  }
